File: scripts/update_versions_mapping.py
from typing import List, Dict
import sys
import json


def main(latest_version: str, json_path: str) -> int:
    with open(json_path, 'r') as f:
        known_versions: Dict[str, str] = json.load(f)
    if latest_version in known_versions.keys():
        return 0

    with open(json_path, 'w') as f:
        json.dump({
            latest_version: latest_version,
            **known_versions,
        }, f, indent=2)
    return 1


if __name__ == '__main__':
    print(main(sys.argv[1].strip(), sys.argv[2]))

File: scripts/update_versions_mapping.mjs
import {readFile, writeFile} from 'fs/promises';
import _ from 'lodash';

async function main() {
  const [latestVersion, jsonPath] = process.argv.slice(2);
  if (!latestVersion) {
    throw new Error(
      'The latest Chromedriver version must be provided as the first command line argument'
    );
  }
  if (!jsonPath) {
    throw new Error(
      'The path to the versions mapping json must be provided as the second command line argument'
    );
  }

  const json = JSON.parse(await readFile(jsonPath, 'utf8'));
  if (latestVersion in json) {
    process.stdout.write('0');
    return;
  }

  const pairs = _.toPairs(json);
  pairs.unshift([latestVersion, latestVersion]);
  await writeFile(jsonPath, JSON.stringify(_.fromPairs(pairs), null, 2), 'utf8');
  process.stdout.write('1');
}

await main();

